src/pages/DownloadMonitor.vue

```diff
@@ -19,14 +19,28 @@
                         <div class="border-at-bottom pad pad--sides">
                             <div class="card is-shadowless">
                                 <p><strong>{{ downloadObject.initialMods.join(", ") }}</strong></p>
-                                <p v-if="downloadObject.progress < 100">Downloading: {{ downloadObject.modName }}</p>
-                                <p>{{Math.min(Math.floor(downloadObject.progress), 100)}}% complete</p>
-                                <Progress v-if="!downloadObject.failed"
-                                    :max='100'
-                                    :value='downloadObject.progress'
-                                    :className="['is-info']"
-                                />
-                                <Progress v-else-if="downloadObject.failed"
+
+                                <div v-if="downloadObject.downloadProgress < 100">
+                                    <p>Downloading: {{ downloadObject.modName }}</p>
+                                    <p>{{(downloadObject.failed ? "Failed at " : "") + Math.min(Math.floor(downloadObject.downloadProgress), 100)}}% complete</p>
+                                    <Progress v-if="!downloadObject.failed"
+                                        :max='100'
+                                        :value='downloadObject.downloadProgress'
+                                        :className="['is-info']"
+                                    />
+                                </div>
+
+                                <div v-else>
```

Comment:
![image](https://github.com/user-attachments/assets/7f9f84b3-27d5-487b-855d-e87b80ece2fc)

State fails when installation throws an error. It's worth checking if errors thrown from other steps as well might cause issues.

![image](https://github.com/user-attachments/assets/df8c5647-6ab2-4ca2-af04-6483db931c83)

Installing the same mod twice causes the state go awry in some fashion. I know it's a niche case but if there's an issue like this it might happen in other scenarios as well so better to figure out what's going on.